Signal/src/ViewControllers/HomeViewController.m

```diff
@@ -510,15 +510,18 @@ - (void)applicationDidBecomeActive:(NSNotification *)notification
             });
         }];
     }
-}
-
-- (void)viewDidAppear:(BOOL)animated
-{
-    [super viewDidAppear:animated];
-
-    if (!self.viewHasEverAppeared) {
-        self.viewHasEverAppeared = YES;
-        [self displayAnyUnseenUpgradeExperience];
+    
+    // We want to show the user the upgrade experience as soon as the app is visible to them.
+    // It cannot go in viewDidAppear, which is called while the app is in the background if
+    // we were launched from a voip notification.
+    if (!self.hasShownAnyUnseenUpgradeExperiences) {
+        dispatch_async(dispatch_get_main_queue(), ^{
+            if ([[UIApplication sharedApplication] applicationState] != UIApplicationStateForeground) {
```

Comment:
fixed this typo in https://github.com/signalapp/Signal-iOS/commit/4d65695bd1e372f48daa0b12fbd013300f33f243 😬